rule.proto

```diff
@@ -17,7 +17,7 @@ message Rule {
   // required
   optional Action action = 3;
 
-  // required
+  // optional
```

Comment:
You can remove the optional comment if the field is optional. The idea is use 
```
// required
optional field = 1
```
to replace 
```
required field = 1
```
Here is the comment i received from Ruben regarding this:
> required has fallen out of favor b/c its not backwards compatible. consider optional instead

Reply:
Oh gotcha. As Ruben pointed out, for this first iteration, I'm not implementing the conditions, so I just removed it altogether.